### PR TITLE
WIP - Trying to remove some of the clutter from Lazyflow

### PR DESCRIPTION
**Pull request by Tomaz-Vieira**
Mostly moving code around and removing weird, probably deprecated stuff. I have no idea if this is stable yet.

I'm also trying to review the locking logic in lazyflow, because it's crashing the GUI since slots might try to service requests when unready because of some disconnect event somewhere.

Do let me know if you see something that was nuked but shouldn't, or something that wasn't and should.

Also, if anyone sees this and knows the answer: is there any good reason to have sub-slots have their parent slot (i.e. the slot just above them, with level == subslot.level + 1) acting like its operator?

### CODE REVIEW

File: lazyflow/slot.py [hunk 1]
```diff
@@ -483,143 +490,71 @@ def _handleUpstreamUnready(self, slot):
...
-                if self._type == upstream_slot._type == "output":
-                    upstream_slot.notifyValueChanged(self._sig_value_changed)
+        if self.upstream_slot is upstream_slot and upstream_slot.level == self.level:
+            return
 
-        except:
-            try:
-                raise
-            finally:
-                try:
```
Comment (emilmelnikov) [inline lazyflow/slot.py]: Have you figured out what was going on here? I was scared of touching that part...

**Comment by k-dominik** [comment]
seems like this has been stale for a while and never made it out of "wip": @Tomaz-Vieira, do you want to get back to it, or scratch it?

**Comment by Tomaz-Vieira** [comment]
> seems like this has been stale for a while and never made it out of "wip": @Tomaz-Vieira, do you want to get back to it, or scratch it?
Nah, we can scratch it. Some of it has already been tackled by maxim anyway =)

**Comment by Tomaz-Vieira** [comment]
Scratched.

**Pull request closed without merging**